tests/unit/test_self_tools.py

```
@@ -95,6 +95,434 @@ def test_remove_context_invalid_prompt():
+        ]
+
+        with patch("bots.tools.self_tools._get_calling_bot", return_value=bot):
+            remove_context("remove math questions")
+
+    # Now try to add a new message - this should work without errors
+    try:
+        user3 = bot.conversation._add_reply(role="user", content="Tell me more")
+        assistant4 = user3._add_reply(role="assistant", content="Sure thing")
+        bot.conversation = assistant4
+        success = True
+    except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)